ArduCopter/compassmot.cpp

```diff
@@ -110,7 +114,7 @@ MAV_RESULT Copter::mavlink_compassmot(const GCS_MAVLINK &gcs_chan)
     }
 
     // get initial compass readings
-    compass.read();
+    bool compass_read_success = compass.read();
```

Comment:
If this fails how can you continue? You won't have a valid `compass_base` presumably?